vyper/semantics/analysis/utils.py

```diff
@@ -591,6 +591,16 @@ def _check_literal(node: vy_ast.VyperNode) -> bool:
         return True
     elif isinstance(node, (vy_ast.Tuple, vy_ast.List)):
         return all(_check_literal(item) for item in node.elements)
+    elif isinstance(node, vy_ast.Attribute):
+        type_ = get_exact_type_from_node(node)
```

Comment:
hmm, this does some type checking, what if `get_exact_type_from_node` is not able to work here? like would `(x: ... = self.x)` be able to resolve?

Reply:
I don't think `def foo(x: ... = self.x)` is currently valid since `self.x` would have `is_constant=False`.

For example, this contract fails compilation:
```
a: uint256

@external
def foo(x: uint256 = self.a):
    pass
```
```
vyper.exceptions.StateAccessViolation: Value must be literal or environment variable
  contract "contracts/env_kwarg.vy:6", function "foo", line 6:21 
       5 @external
  ---> 6 def foo(x: uint256 = self.a):
  ----------------------------^
       7     pass
```